src/huggingface_hub/file_download.py

```diff
@@ -328,14 +330,21 @@ def cached_download(
         )
     }
     if isinstance(use_auth_token, str):
-        headers["authorization"] = f"Bearer {use_auth_token}"
+        auth = cast(
+            AuthHeaders,
+            auth_header(token=use_auth_token, override=override_auth_header),
+        )
+        headers["authorization"] = auth["authorization"]
```

Comment:
Here `auth` can end up being `None` if `auth_header` returns None right ?

In this case `auth["authorization"]` can fail.

It would also be more pythonic to have something like this:
```suggestion
        auth = auth_header(token=use_auth_token, override=override_auth_header)
        if auth:
            headers["authorization"] = auth["authorization"]
```